src/main/java/io/specto/hoverfly/junit/core/Hoverfly.java

```diff
@@ -111,12 +108,12 @@ public void start() {
             setDestination(hoverflyConfig.getDestination());
         }
 
-        if (hoverflyMode == HoverflyMode.CAPTURE) {
-            setMode(hoverflyMode);
-        }
+        setMode(hoverflyMode);
 
-        if (useDefaultSslCert) {
-            sslConfigurer.setTrustStore();
+        if (hoverflyConfig.getAuthenticationConfig().isPresent() && hoverflyConfig.getAuthenticationConfig().get().getSslCert() != null) {
+          sslConfigurer.setDefaultSslContext(hoverflyConfig.getAuthenticationConfig().get().getSslCert());
+        } else if (useDefaultSslCert) {
```

Comment:
When it is false, the user is providing custom ssl to start hoverfly, this was done in the startHoverflyProcess().